movement/io/nwb.py

```diff
@@ -0,0 +1,213 @@
+            description=description,
+            source_software=ds.source_software,
+            skeleton=skeleton_list[-1],
+            **pose_estimation_kwargs,
+        )
+    ]
+
+    skeletons = ndx_pose.Skeletons(skeletons=skeleton_list)
+
+    return pose_estimation, skeletons
+
+
+def convert_movement_to_nwb(
+    nwbfiles: Union[list[pynwb.NWBFile], pynwb.NWBFile],
+    ds: xr.Dataset,
+    pose_estimation_series_kwargs: Optional[dict] = None,
+    pose_estimation_kwargs: Optional[dict] = None,
+    skeletons_kwargs: Optional[dict] = None,
+):
+    if isinstance(nwbfiles, pynwb.NWBFile):
+        nwbfiles = [nwbfiles]
+
+    if len(nwbfiles) != len(ds.individuals):
+        raise ValueError(
+            "Number of NWBFiles must be equal to the number of individuals. "
+            "NWB requires one file per individual."
+        )
+
+    for nwbfile, subject in zip(nwbfiles, ds.individuals.to_numpy()):
+        pose_estimation, skeletons = _create_pose_and_skeleton_objects(
+            ds.sel(individuals=subject),
+            subject,
+            pose_estimation_series_kwargs,
+            pose_estimation_kwargs,
+            skeletons_kwargs,
+        )
+        try:
+            behavior_pm = nwbfile.create_processing_module(
+                name="behavior",
+                description="processed behavioral data",
+            )
+        except ValueError:
+            print("Behavior processing module already exists. Skipping...")
+            behavior_pm = nwbfile.processing["behavior"]
+
+        try:
+            behavior_pm.add(skeletons)
+        except ValueError:
+            print("Skeletons already exists. Skipping...")
+        try:
+            behavior_pm.add(pose_estimation)
+        except ValueError:
+            print("PoseEstimation already exists. Skipping...")
+
+
+def _convert_pse(
+    pes: ndx_pose.PoseEstimationSeries,
+    keypoint: str,
+    subject_name: str,
+    source_software: str,
+    source_file: Optional[str] = None,
+):
+    attrs = {
+        "fps": int(np.median(1 / np.diff(pes.timestamps))),
+        "time_units": pes.timestamps_unit,
+        "source_software": source_software,
+        "source_file": source_file,
+    }
+    n_space_dims = pes.data.shape[1]
+    space_dims = ["x", "y", "z"]
+
+    return xr.Dataset(
+        data_vars={
+            "position": (
+                ["time", "individuals", "keypoints", "space"],
+                pes.data[:, np.newaxis, np.newaxis, :],
+            ),
+            "confidence": (
+                ["time", "individuals", "keypoints"],
+                pes.confidence[:, np.newaxis, np.newaxis],
+            ),
+        },
+        coords={
+            "time": pes.timestamps,
+            "individuals": [subject_name],
+            "keypoints": [keypoint],
+            "space": space_dims[:n_space_dims],
+        },
+        attrs=attrs,
+    )
+
+
+def convert_nwb_to_movement(nwb_filepaths: list[str]) -> xr.Dataset:
```

Comment:
I think my reasoning for not writing from movement dataset to NWB file on disk is because there are often times where you'd like to add other things to the NWB file before writing to disk.